machine/dell/dell_s4000_c2338/i2ctools/i2ctools-sys-eeprom-platform.patch

```diff
@@ -0,0 +1,255 @@
+Add platform specific stuff for the sys-eeprom
+
+diff --git a/sys_eeprom/onie_tlvinfo.c b/sys_eeprom/onie_tlvinfo.c
+index 1ef477c..e63279b 100644
+--- a/sys_eeprom/onie_tlvinfo.c
++++ b/sys_eeprom/onie_tlvinfo.c
+@@ -259,6 +259,7 @@ static void decode_tlv_value(tlvinfo_tlv_t * tlv, char* value)
+     case TLV_CODE_VENDOR_NAME:
+     case TLV_CODE_DIAG_VERSION:
+     case TLV_CODE_SERVICE_TAG:
++    case TLV_CODE_VENDOR_EXT:
+ 	memcpy(value, tlv->value, tlv->length);
+ 	value[tlv->length] = 0;
+ 	break;
+@@ -273,13 +274,6 @@ static void decode_tlv_value(tlvinfo_tlv_t * tlv, char* value)
+     case TLV_CODE_MAC_SIZE:
+ 	sprintf(value, "%u", (tlv->value[0] << 8) | tlv->value[1]);
+ 	break;
+-    case TLV_CODE_VENDOR_EXT:
+-	value[0] = 0;
+-	for (i = 0; (i < (TLV_DECODE_VALUE_MAX_LEN/5)) && (i < tlv->length);
+-	     i++) {
+-	    sprintf(value, "%s 0x%02X", value, tlv->value[i]);
+-	}
+-	break;
```

Comment:
You can't redefine the meaning of the VENDOR_EXT TLV.  It is defined here:

https://opencomputeproject.github.io/onie/docs/design-spec/switch_hw_requirements.html#definition-of-the-tlvinfo-eeprom-format

The reason you can't re-define it's meaning is because then *every* NOS will have to do something special for your platform to decode the VENDOR_EXT.  The existing NOS's have already implemented the ONIE TLV EEPROM format decoder and would have to change *only* for your platform.